### ISSUE TITLE
Compiled index_copy_ writes nothing when the destination has one row

### ISSUE THREAD

**Issue by tdoublep** [torch-spyre/torch-spyre#4007]
A compiled `index_copy_` writes nothing when the destination has a single row. No error is raised. The same code is correct with 8 rows.

torch-spyre: `a31289852145a59099edccc3e506cf5336e8e2e0`

### Reproducer

```python
import torch

def store(out, index, src):
    out.index_copy_(0, index, src)

for rows in (1, 8):
    out = torch.zeros(rows, 8, 128, dtype=torch.float16, device="spyre")
    src = torch.randn(rows, 8, 128, dtype=torch.float16).to("spyre")
    idx = torch.arange(rows, dtype=torch.int32).to("spyre")
    torch.compile(store, dynamic=False)(out, idx, src)
    print(f"rows={rows}: destination still all zero? {bool(out.cpu().eq(0).all())}")
```

Output:

```
rows=1: destination still all zero? True
rows=8: destination still all zero? False
```

Both should be `False`.

Notes:

- Pinning a row-outermost destination layout does not help.
- A compiled `copy_` is correct at one row, so that is a usable workaround where the shapes allow it.
- This may be the part of #3705 that #3882 deliberately left out (the remaining "cannot enforce" early returns). #3705 is closed, so filing separately.

**Comment by tanushree-ibm** [comment]
I am able to reproduce the issue. Currently working on the Fix.

cc: @pradghos @shruthip-venkatesh

**Comment by tanushree-ibm** [comment]
I am working on the fix. PR is in review.

cc : @pradghos @shruthip-venkatesh

**Comment by tanushree-ibm** [comment]
Here is the Root Cause of the above Issue:

Issue: rows=1 Failed (Output Remained All Zeros)
In PyTorch / Inductor:

1. When rows=1, the loop iteration / indexing over idx = torch.arange(1) degenerates to a single index (idx[0] = 0). During loop normalization, expression simplification, and simplify_op_spec(), the dynamic gather/scatter index coordinate was simplified away or folded into a constant offset 0.

2. Because the index expression was folded away, the index tensor idx was no longer referenced by the finalized operation specification (OpSpec), so it was omitted from self.spyre_kernel_args.

3. The bug before the commit(173fc771c4a2d52ff5460254ff252573205cf128): [https://github.com/torch-spyre/torch-spyre/pull/4293/](https://github.com/torch-spyre/torch-spyre/pull/4293/):
Inductor's Python wrapper codegen (call_kernel()) was using all arguments recorded at initial load/store time (python_argdefs()). The wrapper emitted:
_**kernel.run(out, idx, src)**_
However, inside the compiled Spyre kernel / SuperDSC OpSpecs, because idx was folded away, self.spyre_kernel_args only had 2 tensor entries (out as arg 0 and src as arg 1).

- The kernel runtime assigned:
             Arg 0 (out) → Address offset of argument 0 (out)
             Arg 1 (src) → Address offset of argument 1 (idx in the .run() call positional arguments!)

- As a result, the kernel read input data from idx's memory buffer instead of src (or addresses were shifted/misaligned), causing invalid data writes, out-of-bounds writes, or leaving destination buffers unchanged (all zero).

4. In contrast, for rows=8, the index dimension was dynamic/multi-element, so the index tensor idx was not folded away. The argument count in .run() matched the kernel's internal arg_index layout, so rows=8 wrote non-zero values as expected.

**Comment by tanushree-ibm** [comment]
**How PR :** [https://github.com/torch-spyre/torch-spyre/pull/4293/](https://github.com/torch-spyre/torch-spyre/pull/4293/) **solve this**

This commit synchronizes the wrapper's positional arguments in kernel.run(...) with the live arguments surviving in self.spyre_kernel_args:

1. Filters Out Dead Arguments in Codegen:

In SpyreKernel.codegen_kernel():

```
live_names = {name for name, _ in self.spyre_kernel_args}
actuals = []
seen_actuals: set[str] = set()
for name in self.args.python_argdefs()[1]:
    if name in live_names and name not in seen_actuals:
        seen_actuals.add(name)
        actuals.append(name)
self._live_call_arg_names = actuals
```

When idx is folded away for rows=1, live_names only contains {'out', 'src'}. actuals becomes ['out', 'src'].

2. Aligns Kernel Argument Indices with Positional Run Arguments:
    The arg_index assigned to each tensor in OpSpecs (tensor_arg.arg_index = actuals.index(name)) is indexed into actuals.

3. Emits Matching .run() Call in the Wrapper:
    In SpyreKernel.call_kernel():

```
call_args.extend(self._live_call_arg_names)
wrapper.writeline(f"{name}.run({call_args_str})")
```

Instead of emitting kernel.run(out, idx, src), it emits kernel.run(out, src).

With positional arguments and arg_index mappings in exact agreement, the kernel correctly reads from src and writes to out, resolving the rows=1 all-zero issue.

cc : @pradghos @shruthip-venkatesh @joyalbin
